src/layers/layer-factory.android.ts

```diff
@@ -0,0 +1,339 @@
+            circleProperties.push(com.mapbox.mapboxsdk.style.layers.PropertyFactory.circleBlur(new java.lang.Float(propertiesObject['circle-blur'])));
+        }
+
+        if (propertiesObject['circle-color']) {
+            circleProperties.push(com.mapbox.mapboxsdk.style.layers.PropertyFactory.circleColor(propertiesObject['circle-color']));
+        }
+
+        if (propertiesObject['circle-opacity']) {
+            circleProperties.push(com.mapbox.mapboxsdk.style.layers.PropertyFactory.circleOpacity(new java.lang.Float(propertiesObject['circle-opacity'])));
+        }
+
+        if (propertiesObject['circle-radius']) {
+            // we have two options for a radius. We might have a fixed float or an expression
+
+            if (typeof propertiesObject['circle-radius'] == 'number') {
+                circleProperties.push(com.mapbox.mapboxsdk.style.layers.PropertyFactory.circleRadius(new java.lang.Float(propertiesObject['circle-radius'])));
+            } else {
+                if (!propertiesObject['circle-radius'].stops) {
+                    throw new Error('No radius or stops provided to addCircleLayer.');
+                }
+
+                const stopArgs = [];
+
+                for (let i = 0; i < propertiesObject['circle-radius'].stops.length; i++) {
+                    const stop = propertiesObject['circle-radius'].stops[i];
+                    stopArgs.push(com.mapbox.mapboxsdk.style.expressions.Expression.stop(new java.lang.Float(stop[0]), new java.lang.Float(stop[1])));
+                }
+
+                let base = 2;
+
+                if (propertiesObject['circle-radius'].stops.base) {
+                    base = propertiesObject['circle-radius'].stops.base;
+                }
+
+                circleProperties.push(
+                    com.mapbox.mapboxsdk.style.layers.PropertyFactory.circleRadius(
+                        com.mapbox.mapboxsdk.style.expressions.Expression.interpolate(
+                            com.mapbox.mapboxsdk.style.expressions.Expression.exponential(new java.lang.Float(base)),
+                            com.mapbox.mapboxsdk.style.expressions.Expression.zoom(),
+                            stopArgs
+                        )
+                    )
+                );
+            }
+        }
+
+        if (propertiesObject['circle-stroke-color']) {
+            circleProperties.push(com.mapbox.mapboxsdk.style.layers.PropertyFactory.circleStrokeColor(propertiesObject['circle-stroke-color']));
+        }
+
+        if (propertiesObject['circle-stroke-opacity']) {
+            circleProperties.push(com.mapbox.mapboxsdk.style.layers.PropertyFactory.circleStrokeOpacity(new java.lang.Float(propertiesObject['circle-stroke-opacity'])));
+        }
+
+        if (propertiesObject['circle-stroke-width']) {
+            circleProperties.push(com.mapbox.mapboxsdk.style.layers.PropertyFactory.circleStrokeWidth(new java.lang.Float(propertiesObject['circle-stroke-width'])));
```

Comment:
Would be nice to do something like that at the start of the function:
```
const PropertyFactory = com.mapbox.mapboxsdk.style.layers.PropertyFactory;
```
And then use that. It would be faster seeing how the runtime works.
We could do that with other namespaces